components/layout/LanguageSelector.js

```diff
@@ -0,0 +1,53 @@
+import React, { useState, useContext, useEffect } from 'react';
+import PageContext from '../../contexts/PageContext';
+import Router, { useRouter } from 'next/router';
+
+const LanguageSelector = () => {
+    const router = useRouter();
+    const [language, setLanguage] = useState('');
+    const { languageProps } = useContext(PageContext);
+
+    useEffect(() => {
+        setLanguage(languageProps.selectedLanguage);
```

Comment:
What happen if we don't have any language selected (never selected one so no localstorage) and we go to a `/es` prefixed route manually?

Reply:
Spanish content is shown: `yarn start` → http://localhost:3000/es/destinations/costa-rica